internal/report/github/summary.go

```diff
@@ -0,0 +1,127 @@
+package github
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/saucelabs/saucectl/internal/report"
+)
+
+type Reporter struct {
+	beginTime       time.Time
+	stepSummaryFile string
+	results         []report.TestResult
+}
+
+func NewJobSummaryReporter() Reporter {
+	fmt.Printf("GITHUB_STEP_SUMMARY: %s", os.Getenv("GITHUB_STEP_SUMMARY"))
+	return Reporter{
+		beginTime:       time.Now(),
+		stepSummaryFile: os.Getenv("GITHUB_STEP_SUMMARY"),
+	}
+}
+
+func (r *Reporter) isActive() bool {
+	return r.stepSummaryFile != ""
+}
+
+func (r *Reporter) Add(t report.TestResult) {
+	if !r.isActive() {
+		return
+	}
+	r.results = append(r.results, t)
+}
+
+func hasSomeDevice(results []report.TestResult) bool {
```

Comment:
Conveys the same information, but shorter:
```suggestion
func hasDevice(results []report.TestResult) bool {
```